### ISSUE TITLE
Ensure server examples are up to date

### ISSUE THREAD

**Issue by deontologician** [rethinkdb/horizon#425]
#394 indicated some of the options passed to the express server were out of date. It's reasonable to think the koa and hapi examples might be incorrect now as well.

**Comment by Rajan** [comment]
I can confirm that koa example isn't working.

`Rajans-MacBook-Pro:koa-server rajanchandi$ node --harmony main.js 
usage: main.js [-h]

```
           {init,serve,version,create-cert,get-schema,set-schema,make-token}
           ...
```

main.js: error: too few arguments`

**Comment by zigomir** [comment]
I can confirm that I struggled a bit to to get up and running with server examples.
Being pretty new to `RethinkDB` ecosystem, I needed to first figured out how to run Express example.

To get it running I first needed to see what options server is accepting and I can now run it with this:

``` js
const express = require('express');
const horizon = require('@horizon/server');

const app = express();
const http_server = app.listen(8181);
const options = {
  auth: {
    token_secret: 'my_super_secret_secret'
  },
  // Dev like options
  project_name: 'express_server',
  auto_create_collection: true,
  auto_create_index: true,
  permissions: false
};
const horizon_server = horizon(http_server, options);
```

Where you need to run `./main.js` to get Express up and running, and you of course need to run `rethinkdb` in background or in another terminal tab.

Now what I'm most interested in is what do I do with `horizon_server` or `horizon` here?  I guess it's not the same thing and API as on a client? How do I use now RethinkDB from Express server?

I'd be glad to contribute to examples/guides but first need to figure it out myself. I think it would make sense to put server examples in different directory than client examples.

I'm mostly interested in using backend since GeoLocation isn't yet implemented in horizon and I thought maybe you can get to it with extending the server. Or is it actually easier to run separate express + node for it?

**Comment by download13** [comment]
Knowing how to add auth providers would be helpful too.

```
const appServer = app.listen(80, () =>
    console.log('Listening' + (DEV ? ' dev' : ''))
);


const horizonServer = Horizon(appServer);
horizonServer.add_auth_provider(twitterProvider);
```

Is there a place I should get twitter provider? Should I import it from inside the server module?

**Comment by courthead** [comment]
I agree that there should be more documentation on how to use all the server features. I think it will be a common use-case to run both express and horizon together.

@download13 I just did it like this and it seems to work:

``` es6
const googleProvider = require('../node_modules/@horizon/server/src/auth/google');
const horizon = require('@horizon/server');

// `httpsServer` is a server created via the `https` package
const horizonServer = horizon(httpsServer, {
  auth: { token_secret: uuid.v4(), }, // I'm not really sure what to put for the token secret right now
});

horizonServer.add_auth_provider(googleProvider, {
  id: 'google client ID here',
  path: 'not sure what this is for, exactly... I put the path of my authorized redirect URI',
  secret: 'google client secret here',
});
```

**Comment by petermoresi** [comment]
+1 For better documentation. I attempted to integrate the koa example into an existing project to no avail.

**Comment by danielmewes** [comment]
Related documentation issue: https://github.com/rethinkdb/horizon-docs/issues/77
